src/main/scala/beam/agentsim/infrastructure/ScaleUpCharging.scala

```diff
@@ -259,30 +299,23 @@ trait ScaleUpCharging extends {
       } else if (vehicle.isSharedVehicle) {
         VehicleManager.getReservedFor(vehicle.vehicleManagerId.get()).getOrElse(VehicleManager.AnyManager)
       } else VehicleManager.AnyManager
-      val estimatedChargingDuration = Math.max(
+      val estimatedParkingDuration = Math.max(
         inquiry.parkingDuration.toInt,
         beamConfig.beam.agentsim.agents.parking.estimatedMinParkingDurationInSeconds.toInt
       )
-      val (durationToCharge, energyToCharge) =
-        vehicle.refuelingSessionDurationAndEnergyInJoulesForStall(
-          Some(stall),
-          sessionDurationLimit = Some(estimatedChargingDuration),
-          stateOfChargeLimit = None,
-          chargingPowerLimit = None
-        )
+      val remainingRangeInMeters = vehicle.getRemainingRange._1
+      val activityType =
+        if (inquiry.activityType.startsWith(ChargingNetwork.EnRouteLabel))
+          ParkingActivityType.Charge.entryName
+        else inquiry.activityType
       vehicleRequests.synchronized {
```

Comment:
`synchronized` is not needed because this structure is used within `Actor.receive` method.

Reply:
Okay